src/main.cpp

```diff
@@ -132,52 +154,132 @@ void DDWAF::dispose(const Napi::CallbackInfo& info) {
+
   ddwaf_object diagnostics;
 
-  mlog("Update DDWAF instance");
-  ddwaf_handle updated_handle = ddwaf_update(this->_handle, &update, &diagnostics);
-  ddwaf_object_free(&update);
+  mlog("Applying new config to builder");
+  bool update_result = ddwaf_builder_add_or_update_config(this->_builder, LSTRARG(config_path), &update, &diagnostics);
 
   Napi::Value diagnostics_js = from_ddwaf_object(&diagnostics, env);
   info.This().As<Napi::Object>().Set("diagnostics", diagnostics_js);
 
   ddwaf_object_free(&diagnostics);
 
-  if (updated_handle == nullptr) {
-    mlog("DDWAF updated handle is null");
-    Napi::Error::New(env, "WAF has not been updated").ThrowAsJavaScriptException();
-    return;
+  if (!update_result) {
+    mlog("DDWAF Builder update config has failed");
+    return Napi::Boolean::New(env, false);
   }
 
-  mlog("New DDWAF updated instance")
-  ddwaf_destroy(this->_handle);
-  this->_handle = updated_handle;
+  mlog("Update DDWAF instance");
+  ddwaf_handle updated_handle = ddwaf_builder_build_instance(this->_builder);
+  ddwaf_object_free(&update);
 
-  this->update_known_addresses(info);
-  this->update_known_actions(info);
+  if (updated_handle != nullptr) {
+    mlog("New DDWAF updated instance")
+    ddwaf_destroy(this->_handle);
+    this->_handle = updated_handle;
+
+    this->update_known_addresses(info);
+    this->update_known_actions(info);
+  }
+
+  return Napi::Boolean::New(env, true);
+}
+
+Napi::Value DDWAF::remove_config(const Napi::CallbackInfo& info) {
+  mlog("Calling remove config on DDWAF");
+
+  Napi::Env env = info.Env();
+
+  if (this->_disposed) {
+    Napi::Error::New(env, "Could not update a disposed WAF instance").ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+
+  if (info.Length() < 1) {
+    Napi::Error::New(env, "Wrong number of arguments, expected at least 1").ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+
+  if (!info[0].IsString()) {
+    Napi::TypeError::New(env, "First argument must be a string").ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+
+  mlog("Obtaining config remove path");
+  std::string config_path_str = info[0].As<Napi::String>().Utf8Value();
+  const char* config_path = config_path_str.c_str();
+
+  mlog("Applying removed config to builder");
+  bool remove_result = ddwaf_builder_remove_config(this->_builder, LSTRARG(config_path));
+
+  if (!remove_result) {
+    mlog("DDWAF Builder remove config has failed");
+    return Napi::Boolean::New(env, false);
+  }
+
+  mlog("Update DDWAF instance");
+  ddwaf_handle updated_handle = ddwaf_builder_build_instance(this->_builder);
+
+  if (updated_handle != nullptr) {
+    mlog("New DDWAF updated instance")
+    ddwaf_destroy(this->_handle);
+    this->_handle = updated_handle;
+
+    this->update_known_addresses(info);
+    this->update_known_actions(info);
+  }
+
+  return Napi::Boolean::New(env, true);
+}
+
+Napi::Value DDWAF::GetConfigPaths(const Napi::CallbackInfo& info) {
```

Comment:
how are you planning on using this in the library ? if this is going to be read very often, performance wise it might be better to have it as a set we update instead of a getter, to save on the js->cpp perf wall.

Reply:
It will only be called when applying new configurations in RC updates, so I wasn't worried about performance.